incubator/command-dump/src/main/java/io/aklivity/zilla/runtime/command/dump/internal/airline/ZillaDumpCommand.java

```diff
@@ -589,9 +607,19 @@ private void patchExtension(
         private int calculateLabelCrc(
             int labelId)
         {
-            crc.reset();
-            crc.update(resolveLabelAsBytes(labelId));
-            return (int) crc.getValue();
+            int result = 0;
+            if (labelId != 0)
+            {
+                result = crcCache.get(labelId);
+                if (result == 0)
+                {
+                    crc.reset();
+                    crc.update(resolveLabelAsBytes(labelId));
+                    result = (int) crc.getValue();
+                    crcCache.put(labelId, result);
+                }
+            }
+            return result;
```

Comment:
Rename this method as `supplyLabelCrc`, then hoist the code inside `if result == 0` to a separate method called `computeLabelCrc`, then use `result = crcCache.computeIfAbsent(labelId, this::computeLabelCrc);`.

Reply:
OK